doc/specs/#1564 - Settings UI/spec.md

```diff
@@ -0,0 +1,123 @@
+
+#### Proposal 2 - Launch in a new tab
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new tab. This helps us take steps toward supporting non-terminal content in a tab. However, tab tear-off should be implemented before we have this. This way, users can still see their changes take effect if they want by tearing out the settings UI.
+
+### Editing and saving settings
+
+#### Proposal 1 - Automatically save settings
+
+As users edit fields in the settings UI, they are automatically saved and written to the JSON file. This allows the user to see their settings changes taking place in real time.
+
+#### Proposal 2 - Implement a save button
+
+Users will only see their settings changes take place once they click "Save". Clicking "Save" will write to the settings.json file. This aligns with the functionality that exists today by editing the settings.json file in a text editor and saving it.
+
+A future option could be adding a TerminalControl inside the settings UI to preview what the changes will look like before actually saving them to the settings.json file.
+
+## UI/UX Design
+
+### Top-level navigation
+
+#### Proposal 1 - Align with JSON
+
+The settings UI could have top-level navigation that aligns with the overall structure of the settings.json file. The following are the proposed navigation items:
+
+- Globals
+- Profiles
+- Color schemes
+- Bindings
+
+For Bindings, it would have key bindings, mouse bindings, and command palette inside it. I would like to discuss a better name aside from Bindings. I'm afraid having command palette under Bindings would be confusing.
+
+![Settings UI navigation 1](./navigation.png)
+
+#### Proposal 2 - More descriptive navigation
+
+Alternatively, we could break up the navigation menu into more digestible sections. This aligns more closely to other terminals. The following are the proposed navigation items:
+
+- General
+- Appearance
+    - Global
+    - Color schemes
+    - Themes
+- Tabs
+- Profiles
+    - Defaults
+    - Enumerate profiles
+    - Add new
+- Keyboard
+- Mouse*
+- Command Palette
+- Marketplace*
+
+\* Mouse and marketplace will be added once they're implemented.
+
+![Settings UI navigation 2](./navigation-2.png)
+
+## Capabilities
+
+### Accessibility
+
+This will have to undergo full accessibility testing because it is a new UI element. All items inside the settings UI should be accessible by a screen reader and the keyboard. Additionally, all of the settings UI will have to be localized.
+
+### Security
+
+This does not impact security.
+
+### Reliability
+
+This will not improve reliability.
+
+### Compatibility
+
+This will change the default experience to open the UI, rather than the JSON file in a text editor. This behavior can be reverted with the setting listed [above](#ability-to-disable-displaying-the-settings-ui).
+
+### Performance, Power, and Efficiency
+
+This does not affect performance, power, nor efficiency.
+
+## Potential Issues

```

Comment:
Do we have any concerns about the way any of the current settings we have will be exposed to the user? Like, are there any where we're worried about what control we use to let the user change the setting?